### PR TITLE
Switch from cmake-format to gersemi for formatting CMake listfiles

### LINKED ISSUE #719: Switch from dead cmake-format project to `gersemi` for CMake formatting
The CMake formatter we use here (and in our other projects) is [cmake-format](https://github.com/cheshirekow/cmake_format), which hasn't been updated in over 3.5 years (last commit in May 2021). It seems the author abandoned it entirely, many PRs are still open and there has been discussion about having a successor fork on some issues e.g. https://github.com/cheshirekow/cmake_format/issues/340

The linked issue suggests using this project: https://github.com/BlankSpruce/gersemi. Here's what I see is good here:
- has pre-commit hooks
- is up-to-date and maintained
- has a configuration file (`.gersemirc`) that pre-commit will pick up
- will format the files in-place

### PR DESCRIPTION
**Pull request by ManifoldFR**
This is a rather large PR (due to reformatting many, many files)

Closes #719

### CODE REVIEW

File: .gersemirc
Comment (nim65s) [inline .gersemirc:7]: Is this necessary ?
Is it going to be necessary on each and every project of ours ? 
Couldn't we just use some default settings instead ?

Comment (ManifoldFR) [reply .gersemirc:7]: I tried to put the options that would create the least changes. I think some of these _might_ be the default values though

Comment (nim65s) [reply .gersemirc:7]: Do we care about the number of changes ? 
Those are already 3k on a 15k codebase, so IMHO it could be 2 or 3 times that and we would not see the difference, as long as it is in a single commit documented in `.git-blame-ignore-revs`

Comment (ManifoldFR) [reply .gersemirc:7]: Okay, in that case, I remove some of the options in the gersemirc file.

**Review by nim65s — APPROVED** [review]
I'm still not sure about that `indent: 2`, but I'm not going to block this PR for that either.
Before merging, I think we should clean git history, with:
1. update .pre-commit-config.yml + setup .gersemirc
2. pre-commit run -a
3. add previous to .git-blame-ignore-revs

**Comment by ManifoldFR** [comment]
> I'm still not sure about that `indent: 2`, but I'm not going to block this PR for that either. Before merging, I think we should clean git history, with:
> 
> 1. update .pre-commit-config.yml + setup .gersemirc
> 2. pre-commit run -a
> 3. add previous to .git-blame-ignore-revs

The default indent in gersemi is 4.
I will edit the git history to create a separate commit for 1. and 2.

**Comment by ManifoldFR** [comment]
Done @nim65s 😃

File: dist.cmake
Comment (jorisv) [inline dist.cmake:62]: Please revert this change and change cmake-format: on/off by gersemi: on/off

File: dist.cmake
Comment (jorisv) [inline dist.cmake:84]: Please revert this change and change cmake-format: on/off by gersemi: on/off

File: dist.cmake
Comment (jorisv) [inline dist.cmake:107]: Please revert this change and change cmake-format: on/off by gersemi: on/off

File: dist.cmake
Comment (jorisv) [inline dist.cmake:130]: Please revert this change and change cmake-format: on/off by gersemi: on/off

File: dist.cmake
Comment (jorisv) [inline dist.cmake:154]: Please revert this change and change cmake-format: on/off by gersemi: on/off

File: dist.cmake
Comment (jorisv) [inline dist.cmake:169]: Please revert this change and change cmake-format: on/off by gersemi: on/off

File: header.cmake
Comment (jorisv) [inline header.cmake:180]: Remove cmake-format here

File: release.cmake
Comment (jorisv) [inline release.cmake:140]: Please revert this change and change cmake-format: on/off by gersemi: on/off

File: release.cmake
Comment (jorisv) [inline release.cmake:199]: Please revert this change and change cmake-format: on/off by gersemi: on/off

**Review by jorisv — CHANGES_REQUESTED** [review]
Be sure to replace all # cmake-format: by # gersemi:

Also, I'm curious on how well gersemi handle custom command: https://github.com/BlankSpruce/gersemi?tab=readme-ov-file#lets-make-a-deal

On Pinocchio with cmake-format, I had to do this: https://github.com/stack-of-tasks/pinocchio/blob/devel/.cmake-format.yaml

Last, did you know if gersemi is handling comment in a better way ? cmake-format was joining consecutive line together, make huge comment block.

**Comment by ManifoldFR** [comment]
> Be sure to replace all # cmake-format: by # gersemi:
> 
> Also, I'm curious on how well gersemi handle custom command: https://github.com/BlankSpruce/gersemi?tab=readme-ov-file#lets-make-a-deal
> 
> On Pinocchio with cmake-format, I had to do this: https://github.com/stack-of-tasks/pinocchio/blob/devel/.cmake-format.yaml
> 
> Last, did you know if gersemi is handling comment in a better way ? cmake-format was joining consecutive line together, make huge comment block.

We can do the custom command stuff later. Checking for this would require some more in-depth look at our own code so that gersemi can pick up definitions, or introduce stubs to tell gersemi how args should be formatted.

**Review by jorisv — APPROVED** [review]

**Pull request merged**
